### DISCUSSION TITLE
New `{{Variable}}` -> {{tags}}

### DISCUSSION THREAD

**Discussion by FelipeRearden** [Taitava/obsidian-shellcommands#50] [Ideas - done & released]
Hello @Taitava !!!!

As we talk before about this plugin [Cross-Reference](https://github.com/alexobenauer/Cross-reference-Navigation-for-Obsidian), I'm starting using tags and I think a have a new idea for a variable `{{tags}}` from the current/active note :)

### My workflow 
When I'm writing I start to type tags in the middle (inside) of the text to not lose momentum when I'm focus. 

My text would look like this. I know, its a mess :(
![Screen Shot 2021-09-27 at 06 45 59](https://user-images.githubusercontent.com/85180696/134888490-b2b68b79-dccf-40c8-94b3-4d99f4dca4a1.png)

My idea is to gahter all tags from a note and transfer to an specific heading in the same note (maybe in the end)

`echo {{tags}} >> "/{{file_path:absolute}}"` or `printf {{tags}} | pbcopy` ( I don't know yet)

Then I use Regex to delete the tags in the text to clean the mess. 

### One Special Request 🙏

When dealing with long texts, it's commom that I write many times the same tag, generating more than one record of the same tag in the some. 

Using `{{tags}}` as a list of tags from a note, I'd like to ask for deleting duplicates records of tags. (one record for each tag) 

Example: this note has the following 4 tags:
#amor 
#ação 
#medo 
#medo 

For the `{{tags}}` is was thinking about this result: 
#amor 
#ação 
#medo 

---

I hope you like this new variable 

I wish you a fantastic Monday!

**Comment by Taitava** [comment]
An excellent idea! 🙂 

> Using {{tags}} as a list of tags from a note, I'd like to ask for deleting duplicates records of tags. (one record for each tag)

Makes sense. I haven't yet used Obsidian's API to get tags, so I don't know how it works, but I'd guess the API can already give a list of tags without duplicates. Nevertheless, duplicates can be removed even if the API would not do it for some reason.

In some database languages (such as [SQL languages](https://en.wikipedia.org/wiki/SQL)), a term *distinctive* is used for getting results without duplicates.

The definition for this variables would be: `{{tags:separator}}`, where *separator* is a freely decideable text that will be used between tags, and it's mandatory.

Have a nice day, too! 🙂

**Reply by FelipeRearden** [reply]
Thank you very much @Taitava !!!!

> Makes sense. I haven't yet used Obsidian's API to get tags, so I don't know how it works, but I'd guess the API can already give a list of tags without duplicates. Nevertheless, duplicates can be removed even if the API would not do it for some reason.

I hope it's an easy one :)

> In some database languages (such as SQL languages), a term distinctive is used for getting results without duplicates.

One more term that I learn with you :)

Thanks again, I'm gonna open an FR for this new feature.

Have a great day!

**Comment by Taitava** [comment]
> I hope it's an easy one :)

Should be. (That's something a programmer should never say. Things that should be easy tend to extend their amount of work from what was expected 😲 ).

Thanks, have a nice day too! 🙂

**Comment by FelipeRearden** [comment]
> Should be. (That's something a programmer should never say. Things that should be easy tend to extend their amount of work from what was expected 😲 ).


It's funny how theory is different from practice: We always have to deal with the unexpected :smiley 

That's why you guys are so important to our world. You have te knowledge to transform an idea a thought into a tool/product/asset that can change people (like me) lives :)

That's your superpower :)

Have a great day @Taitava !

**Reply by Taitava** [reply]
Thanks! 🙂 You are important too, there's always a role for everyone! A tool needs users so that it can be improved through real word usage. 😎  Have a nice day too!

**Reply by FelipeRearden** [reply]
Thanks @Taitava . Let's go !!!!!!!!!!!!!!!!!!!!
